src/triggers/commentDocumentV2.ts

```diff
@@ -0,0 +1,254 @@
+      avatarUrl: string;
+    };
+  } | null;
+}
+
+interface CommentsResponse {
+  data: {
+    comments: {
+      nodes: Comment[];
+    };
+  };
+}
+
+/**
+ * Sets up a new webhook subscription for document comments in Linear.
+ * @see https://platform.zapier.com/build/cli-hook-trigger#1-write-the-subscribehook-function
+ * @see https://platform.zapier.com/build/cli-hook-trigger#subscribehook
+ */
+const subscribeHook = (z: ZObject, bundle: Bundle) => {
+  const data = {
+    url: bundle.targetUrl,
+    inputData:
+      bundle.inputData && Object.keys(bundle.inputData).length > 0
+        ? pick(bundle.inputData, ["creatorId", "projectId", "documentId"])
+        : undefined,
+  };
+
+  return z
+    .request({
+      url: "https://client-api.linear.app/connect/zapier/subscribe/commentDocument",
+      method: "POST",
+      body: data,
+    })
+    .then((response) => response.data);
+};
+
+/**
+ * Deletes a webhook subscription for document comments in Linear.
+ * @see https://platform.zapier.com/build/cli-hook-trigger#2-write-the-unsubscribehook-function
+ * @see https://platform.zapier.com/build/cli-hook-trigger#unsubscribehook
+ */
+const unsubscribeHook = (z: ZObject, bundle: Bundle) => {
+  // bundle.subscribeData contains the parsed response JSON from the subscribe request.
+  const hookId = bundle.subscribeData?.id;
+
+  return z
+    .request({
+      url: `https://client-api.linear.app/connect/zapier/unsubscribe/${hookId}`,
+      method: "DELETE",
+    })
+    .then((response) => response.data);
+};
+
+/**
+ * This processes inbound webhooks from Linear.
+ * @see https://platform.zapier.com/build/cli-hook-trigger#3-write-the-perform-function
+ * @see https://platform.zapier.com/build/cli-hook-trigger#perform
+ */
+const getComment = (z: ZObject, bundle: Bundle) => {
+  const comment = {
+    ...bundle.cleanedRequest.data,
+    querystring: undefined,
+  };
+
+  return [comment];
+};
+
+/**
+ * Fetches a list of comments from Linear to use as examples when building a Zap.
+ * @see https://platform.zapier.com/build/cli-hook-trigger#4-write-the-performlist-function
+ * @see https://platform.zapier.com/build/cli-hook-trigger#performlist
+ */
+const getCommentList = () => async (z: ZObject, bundle: Bundle) => {
+  const variables = omitBy(
+    {
+      creatorId: bundle.inputData.creatorId,
+      projectId: bundle.inputData.projectId,
+      documentId: bundle.inputData.documentId,
+    },
+    (v: undefined) => v === undefined
+  );
+
+  const filters = [];
+  if ("creatorId" in variables) {
+    filters.push(`{ user: { id: { eq: $creatorId } } }`);
+  }
+  if ("projectId" in variables) {
+    filters.push(
+      `{ or: [{ documentContent: { project: { id: { eq: $projectId }} } }, { documentContent: { document: { project: { id: { eq: $projectId }}}}}] }`
+    );
+  }
+  if ("documentId" in variables) {
+    filters.push(`{ documentContent: { document: { id: { eq: $documentId }}}}`);
+  }
+
+  const response = await z.request({
```

Comment:
Is this different to V1, can we not reuse the listing request?

Reply:
It’s slightly different since we don’t need pagination to fetch samples and I also modified the schema slightly (no more `resolvingUser`, no exposing the ID of the document content object directly)